api/billingd/filrewards/filrewards.go

```diff
@@ -0,0 +1,217 @@
+package filrewards
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/textileio/textile/v2/api/billingd/analytics"
+	mdb "github.com/textileio/textile/v2/mongodb"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Event int
+
+const (
+	Unspecified Event = iota
+	FirstKeyAccountCreated
+	FirstKeyUserCreated // does this fit "register first user?"
+	FirstOrgCreated
+	InitialBillingSetup
+	FirstBucketCreated
+	FirstBucketArchiveCreated
+	FirstMailboxCreated
+	FirstThreadDbCreated
+)
+
+const baseAttoFILReward = 1000 // What should this be? Should we read it from mongo?
```

Comment:
This is the amount that will be rewarded for the smallest rewards, and it is multiplied based on the reward data just below.

Reply:
perfect! `baseAttoFILReward` could just be a startup flag on billing for now. might be nice to set a startup check with a max baseAttoFILReward that hard stops